spec-dtslint/operators/tap-spec.ts

```diff
@@ -0,0 +1,24 @@
+import { of } from 'rxjs';
+import { tap } from 'rxjs/operators';
+
+it('should infer correctly', () => {
+  const a = of(1, 2, 3).pipe(tap()); // $ExpectType Observable<number>
+});
+
+it('should accept partial observer', () => {
+  const a = of(1, 2, 3).pipe(tap({ next: (x: number) => { } })); // $ExpectType Observable<number>
+  const b = of(1, 2, 3).pipe(tap({ error: (x: any) => { } })); // $ExpectType Observable<number>
+  const c = of(1, 2, 3).pipe(tap({ complete: () => { } })); // $ExpectType Observable<number>
+});
+
+it('should not accept empty observer', () => {
+  const a = of(1, 2, 3).pipe(tap({})); // $ExpectError
+});
+
+it('should enforce type for next observer function', () => {
+  const a = of(1, 2, 3).pipe(tap({ next: (x: string) => { })); // $ExpectError
+});
+
+it('should accept any type for error observer function', () => {
```

Comment:
This test can be removed, as it's the same as the one in the `should accept partial observer` test.